### PR TITLE
IAP samples update

### PR DESCRIPTION
**Pull request by jabubake**
- switching to nimbus library for ease of JWK loading
- adding constraints for iat, alg

@matthewg

### CODE REVIEW

File: iap/src/main/java/com/example/iap/VerifyIapRequestHeader.java [hunk 1]
```diff
@@ -46,127 +37,93 @@
...
-        PublicKey publicKey = kf.generatePublic(keySpec);
-        if (publicKey instanceof ECPublicKey) {
-          return (ECPublicKey) publicKey;
-        }
-      } catch (InvalidKeySpecException | NoSuchAlgorithmException var7) {
-        throw new IOException("Unexpected exception reading data", var7);
-      }
-    }
-    return null;
+  boolean verifyJwt(String jwtToken, String expectedAudience) throws Exception {
```
Comment (lesv) [inline iap/src/main/java/com/example/iap/VerifyIapRequestHeader.java]: Shouldn't there also be an expires time that should be verified?

**Comment by lesv** [comment]
P.S. Circle was unhappy -- as long as you got tested, I'm Ok.

File: iap/src/main/java/com/example/iap/VerifyIapRequestHeader.java (hunk 1 above)
Comment (jabubake) [inline iap/src/main/java/com/example/iap/VerifyIapRequestHeader.java]: Good catch, added

**Comment by jabubake** [comment]
`mvn clean verify`  did work for me locally.

**Pull request merged**